app/models/ability.rb

```diff
@@ -10,102 +8,110 @@ def initialize(user)
-      user.admin? || on_team?(user, repo.team)
-    end
+      can :crud, Team do |team|
+        user.admin? || signed_in?(user) && team.new_record? || on_team?(user, team)
+      end
 
-    can :supervise, Team do |team|
-      user.roles.organizer.any? || team.supervisors.include?(user)
-    end
+      can :update_conference_preferences, Team do |team|
+        team.accepted? && team.students.include?(user)
+      end
 
-    can :crud, ConferencePreference do |preference|
-      user.admin? || (preference.team.students.include? user)
-    end
+      can :see_offered_conferences, Team do |team|
+        user.admin? || team.students.include?(user) || team.supervisors.include?(user)
+      end
 
-    can :crud, Conference if user.admin? || user.current_student?
+      can :accept_or_reject_conference_offer, Team do |team|
+        team.students.include?(user)
+      end
 
-    # todo add mailing controller and view for users in their namespace, where applicable
-    can :read, Mailing do |mailing|
-      mailing.recipient? user
-    end
+      cannot :create, Team do |team|
+        on_team_for_season?(user, team.season) || !user.confirmed?
+      end
 
-    can :crud, :comments if user.admin?
-    can :read, :users_info if user.admin? || user.supervisor?
+      can :join, Team do |team|
+        team.helpdesk_team? and signed_in?(user) and user.confirmed? and not on_team?(user, team)
+      end
 
-    # projects
-    can :crud, Project do |project|
-      user.admin? ||
-        (user.confirmed? && user == project.submitter)
-    end
-    can :use_as_template, Project do |project|
-      user == project.submitter && !project.season&.current?
-    end
+      can :crud, Role do |role|
+        user.admin? || on_team?(user, role.team)
+      end
 
-    can :create, Project if user.confirmed?
-    cannot :create, Project if !user.confirmed?
+      can :crud, Source do |repo|
+        user.admin? || on_team?(user, repo.team)
+      end
 
-    # activities
-    can :read, :feed_entry
-    can :read, :mailing if signed_in?(user)
+      can :supervise, Team do |team|
+        user.roles.organizer.any? || team.supervisors.include?(user)
+      end
 
-    # applications
-    can :create, :application_draft if user.student? && user.application_drafts.in_current_season.none?
-  end
+      can :crud, ConferencePreference do |preference|
+        user.admin? || (preference.team.students.include? user)
+      end
 
-  def signed_in?(user)
-    user.persisted?
-  end
+      can :crud, Conference if user.admin? || user.current_student?
 
-  def on_team?(user, team)
-    user.teams.include?(team)
-  end
+      # todo add mailing controller and view for users in their namespace, where applicable
+      can :read, Mailing do |mailing|
+        mailing.recipient? user
+      end
 
-  def on_team_for_season?(user, season)
-    season && user.roles.student.joins(:team).pluck(:season_id).include?(season.id)
-  end
+      can :crud, :comments if user.admin?
```

Comment:
This is not related to this PR specifically, and we should probably open a separate issue for this: it seems we're not testing this ability in the project comments (or anywhere I could find).

We used to have this in the teams show page, but it was removed meanwhile: https://github.com/rails-girls-summer-of-code/rgsoc-teams/blob/4fc8dfdec20a377fd3a1a500e2399aac7cea48d5/app/views/teams/show.html.slim#L17-L22

I've not used CanCanCan a lot in the past.. do we usually use the [controller helpers](https://github.com/CanCanCommunity/cancancan#3-controller-helpers) or just view helpers?

Should I create the issue or do we want to fix it in this PR?

Reply:
We use both helpers ( PR #993 is all about the controller helpers :-) )
If it's okay with you, I'd prefer to keep the PR's as clean as possible. And collect issues for every step layed out in the Overhaul: Breakdown (#991) . So we can use that as a reference all the way.
The `crud comments issue` will be part of the CommentsController Overhaul (listed in nr 5 in 991).  Or afterwards, in the restructuring of the Ability. I'll add it. Thanks!!

Reply:
Of course! Thank you for your patience - I'm still figuring out how we structure things! 😊